### ISSUE TITLE
Can't open .targets files

### ISSUE THREAD

**Issue by StephaneDelcroix** [mono/monodevelop#6589]
Trying to open our Xamarin.Forms.targets

<img width="1187" alt="image" src="https://user-images.githubusercontent.com/313003/48435794-44f0df00-e74b-11e8-9ac1-f256614d9230.png">

=== Visual Studio Enterprise 2017 for Mac (Preview) ===

Version 7.7 Preview (7.7 build 1837)
Installation UUID: 432670e0-72c2-49ad-9697-54e2fbd0e94c
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 4.4.1.178 (master / eeaeb7e6)

	Package version: 516000215

=== Mono Framework MDK ===

Runtime:
	Mono 5.16.0.215 (2018-06/eee1d634e8b) (64-bit)
	Package version: 516000215

=== NuGet ===

Version: 4.7.0.5148

=== .NET Core ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	2.1.2
	2.1.1
	2.0.5
	2.0.0
	1.1.1
	1.0.4
SDK: /usr/local/share/dotnet/sdk/2.1.302/Sdks
SDK Versions:
	2.1.302
	2.1.301
	2.1.4
	2.0.0
	1.0.1
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/5.16.0/lib/mono/msbuild/15.0/bin/Sdks

=== Xamarin.Profiler ===

Version: 1.6.4
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Apple Developer Tools ===

Xcode 10.1 (14460.46)
Build 10B61

=== Xamarin.Mac ===

Version: 5.2.1.10 (Visual Studio Enterprise)
Hash: f2a05edd
Branch: 
Build date: 2018-10-31 18:55:58-0400

=== Xamarin.iOS ===

Version: 12.2.1.10 (Visual Studio Enterprise)
Hash: f2a05edd
Branch: d15-9
Build date: 2018-10-31 18:55:57-0400

=== Xamarin.Android ===

Version: 9.1.0.38 (Visual Studio Enterprise)
Android SDK: /Users/sdx/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		4.4.87 (API level 20)
		6.0    (API level 23)
		7.0    (API level 24)
		7.1    (API level 25)
		8.0    (API level 26)
		8.1    (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 28.0.1
SDK Build Tools Version: 28.0.0

Java SDK: /Users/sdx/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1
openjdk version "1.8.0-9"
OpenJDK Runtime Environment (build 1.8.0-9-microsoft-b00)
OpenJDK 64-Bit Server VM (build 25.71-b00, mixed mode)

Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android Device Manager ===

Version: 7.7.1.0
Hash: ee948863

=== Xamarin Inspector ===

Version: 1.4.3
Hash: db27525
Branch: 1.4-release
Build date: Mon, 09 Jul 2018 21:20:18 GMT
Client compatibility: 1

=== Build Information ===

Release ID: 707001837
Git revision: de8219e9847e49d6365742357a05cc74bec22ee5
Build date: 2018-11-12 11:10:55+00
Build branch: release-7.7
Xamarin extensions: 64d562a1e28bed285dce6d33a48ca6fcb75236a4

=== Operating System ===

Mac OS X 10.14.2
Darwin 18.2.0 Darwin Kernel Version 18.2.0
    Sat Nov  3 12:30:49 PDT 2018
    root:xnu-4903.231.1~11/RELEASE_X86_64 x86_64

=== Enabled user installed extensions ===

DotNetCoreSupportAnyTargetFramework 1.0
MSBuild Editor 2.1.0
NuGet Package Management Extensions 0.14
Show in Github 1.0.4

> VS bug [#732827](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/732827)

**Comment by Therzok** [comment]
@StephaneDelcroix could you please attach the log files for this issue? Maybe MSBuild Editor addin is causing the failure. 

Tried to reproduce this locally with Version 7.7 Preview (7.7 build 1861), and could not.

**Comment by Therzok** [comment]
cc @mhutch

**Comment by Therzok** [comment]
Yup,
```
System.ArgumentNullException: Value cannot be null.
Parameter name: key
  at System.Collections.Generic.Dictionary`2[TKey,TValue].TryInsert (TKey key, TValue value, System.Collections.Generic.InsertionBehavior behavior) [0x002b5] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/Collections/Generic/Dictionary.cs:677 
  at System.Collections.Generic.Dictionary`2[TKey,TValue].set_Item (TKey key, TValue value) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/Collections/Generic/Dictionary.cs:231 
  at MonoDevelop.MSBuildEditor.Schema.TaskMetadataBuilder..ctor (MonoDevelop.MSBuildEditor.Language.MSBuildRootDocument rootDoc) [0x00075] in <044162a39db34176ae2613d2015032db>:0 
  at MonoDevelop.MSBuildEditor.Language.MSBuildRootDocument.Parse (System.String filename, MonoDevelop.Core.Text.ITextSource textSource, MonoDevelop.MSBuildEditor.Language.MSBuildRootDocument previous, MonoDevelop.MSBuildEditor.Schema.MSBuildSchemaProvider schemaProvider, MonoDevelop.MSBuildEditor.Language.IRuntimeInformation runtimeInfo, System.Threading.CancellationToken token) [0x00155] in <044162a39db34176ae2613d2015032db>:0 
  at MonoDevelop.MSBuildEditor.MSBuildParsedDocument.ParseInternal (MonoDevelop.Ide.TypeSystem.ParseOptions options, System.Threading.CancellationToken token) [0x00065] in <044162a39db34176ae2613d2015032db>:0 
  at MonoDevelop.MSBuildEditor.MSBuildDocumentParser+<>c__DisplayClass0_0.<Parse>b__0 () [0x00000] in <044162a39db34176ae2613d2015032db>:0 
  at System.Threading.Tasks.Task`1[TResult].InnerInvoke () [0x0000f] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corert/src/System.Private.CoreLib/src/System/Threading/Tasks/Future.cs:544 
  at System.Threading.Tasks.Task.Execute () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corert/src/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:2343 
--- End of stack trace from previous location where exception was thrown ---

  at MonoDevelop.Ide.Editor.TextEditorViewContent.RunFirstTimeFoldUpdate (System.String text) [0x000d4] in /Users/vsts/agent/2.142.1/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/TextEditorViewContent.cs:200 
  at MonoDevelop.Ide.Editor.TextEditorViewContent.Load (MonoDevelop.Ide.Gui.FileOpenInformation fileOpenInformation) [0x000d0] in /Users/vsts/agent/2.142.1/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/TextEditorViewContent.cs:220 
  at MonoDevelop.Ide.Gui.LoadFileWrapper.Invoke (System.String fileName) [0x00165] in /Users/vsts/agent/2.142.1/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Workbench.cs:1614 
```

**Comment by mhutch** [comment]
This is the code. Looks like something is up with one of your projects if it doesn't have an output name.

```csharp
foreach (var project in Ide.IdeApp.Workspace.GetAllProjects ()) {
	var outName = project.GetOutputFileName (project.ParentSolution.DefaultConfigurationSelector);
	projectMap [Path.GetFileNameWithoutExtension (outName)] = project;
}
```

**Comment by StephaneDelcroix** [comment]
still happen on 7.7 build 1841

**Comment by mhutch** [comment]
Is it just in that solution, or in *any* solution?

**Comment by StephaneDelcroix** [comment]
still happening on 1861, on XF sln

**Comment by Therzok** [comment]
@StephaneDelcroix Disable the msbuild editor addin in the meantime to unblock this. It's not from a default VSMac install.

**Comment by Therzok** [comment]
Re-opening the issue as the editor could safe-guard against that.

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-18] Comment by **xamarinc**



Fixed in version 7.8.0.1219 (master)



Pull Request #6637 merged by: Lluis Sanchez  
 Author: mono  
 Commit: b27491d2858260d903eab6ee4987bd022f23c2f9 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-19] Comment by **xamarinc**



Fixed in version 8.0.0.777 (release-8.0)



Pull Request #6637 merged by: Lluis Sanchez  
 Author: mono  
 Commit: b27491d2858260d903eab6ee4987bd022f23c2f9 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-20] Comment by **xamarinc**



Fixed in version 7.8.0.1310 (release-7.8)



Pull Request #6637 merged by: Lluis Sanchez  
 Author: mono  
 Commit: b27491d2858260d903eab6ee4987bd022f23c2f9 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-30] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #6637 merged by: Lluis Sanchez  
 Author: mono  
 Commit: b27491d2858260d903eab6ee4987bd022f23c2f9 (xamarin/monodevelop)
